src/lib/components/BucketAggregation/BucketAggregationValues.js

```diff
@@ -152,7 +158,7 @@ const ValueElement = ({
       <List.Item key={bucket.key}>
         <Checkbox
           label={label}
-          id={`${bucket.key}-agg-value-checkbox`}
+          id={`${parentValue}-${bucket.key}-agg-value-checkbox`}
```

Comment:
question: I assume that this is the fix differentiating the components, can you point me to the code that utilizes this?

Reply:
What I believe is happening (to be confirmed) is that `React` is adding an event listener (click handler) on the `id` of the HTML tag. If you look how it is rendered, there is no inline `onClick` or similar:
`<input class="hidden" id="Validation-agg-value-checkbox" readonly="" tabindex="0" type="checkbox" value="Validation">`

This means that is another checkbox has the same `id`, ids become duplicated and the first is picked.
Having unique ids looks like fixes the issue.

Reply:
ah right, I thought it was somewhat connected with our internal implementation of filtering. It makes sense, thanks for the explanation! Maybe a neat comment @psaiz: 
```suggestion
          id={parentValue ? `${parentValue}-${bucket.key}-agg-value-checkbox` : `${bucket.key}-agg-value-checkbox`}
```

Reply:
Ok, comment added